modules/minileven/minileven.php

```diff
@@ -117,7 +117,9 @@ function jetpack_mobile_template( $theme ) {
 }
 
 function jetpack_mobile_available() {
-	echo '<div class="jetpack-mobile-link" style="text-align:center;margin:10px 0;"><a href="'. home_url( '?ak_action=accept_mobile' ) . '">' . __( 'View Mobile Site', 'jetpack' ) . '</a></div>';
+	global $wp;
+	$current_url =  home_url( add_query_arg( array( 'ak_action' => 'accept_mobile' ), add_query_arg( $_GET, $wp->request ) ) );
```

Comment:
Coming back to this, I'd like to offer a solution that should work regardless of your permalink settings, and even on sites where WP lives in a subdirectory. It's also a bit easier to read as it does not use a `add_query_arg` inside another `add_query_arg`.

```php
	global $wp;
	$url_params = array(
		'ak_action' => 'accept_mobile',
	);
	if ( is_array( $_GET ) && ! empty( $_GET ) ) {
		$url_params[] = $_GET;
	}
	$current_url = home_url(
		add_query_arg(
			$url_params,
			$wp->request
		)
	);

	echo '<div class="jetpack-mobile-link" style="text-align:center;margin:10px 0;"><a href="'. esc_url( $current_url ) . '">' . __( 'View Mobile Site', 'jetpack' ) . '</a></div>';
```

What do you think about this?